api/python/t4/packages.py

```diff
@@ -187,28 +187,49 @@ def get(self):
         """
         return _to_singleton(self.physical_keys)
 
-    def deserialize(self):
+    def deserialize(self, deserializer=None):
         """
         Returns the object this entry corresponds to.
 
+        Custom Deserializers:
+            If `deserializer` is given, it is passed the physical key
+            of the data that would have been deserialized, then its
+            return value is returned.
+
+            Example for decoding a json blob:
+                >>> des = lambda k: json.loads(Path(urlparse(k).path).read_text())
```

Comment:
This is already pretty complicated - but still not enough. E.g., need to `unquote` the `.path`. There's a helper function for this, `parse_file_url` - though it's not really documented as a user API.